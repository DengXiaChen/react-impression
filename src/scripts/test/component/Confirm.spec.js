import React from 'react';
import { shallow } from 'enzyme';
import { Confirm } from '../../components/impression';

const setup = () => {
    const props = {
        onOkClick: jest.fn(),
        onCancelClick: jest.fn(),
        okText: 'ok',
        cancelText: 'cancel',
        type: 'danger',
    };

    const wrapper = shallow(
        <Confirm {...props}>
            Confirm test
        </Confirm>
    );

    return {
        props,
        wrapper,
    };
};

describe('Confirm', () => {
    const { wrapper, props } = setup();

    it('render', () => {
        expect(wrapper.find('.confirm').exists()).toBe(true);
    });

    it('props', () => {
        expect(wrapper.find('.confirm-btn-sure').text()).toBe('ok');
        expect(wrapper.find('.confirm-btn-cancel').text()).toBe('cancel');
        expect(wrapper.find('.text-danger').exists()).toBe(true);
    });

    it('click', () => {
        wrapper.find('.confirm-btn-sure').simulate('click');
        expect(props.onOkClick).toHaveBeenCalled();
        wrapper.find('.confirm-btn-cancel').simulate('click');
        expect(props.onCancelClick).toHaveBeenCalled();
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
