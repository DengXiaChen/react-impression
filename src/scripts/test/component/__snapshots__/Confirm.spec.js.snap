// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="confirm"
>
    <div
        className="confirm-dialog"
    >
        <div
            className="confirm-addon"
        >
            <i
                className="fa fa-exclamation-circle text-danger"
            />
        </div>
        <div
            className="confirm-body"
        >
            Confirm test
        </div>
        <div
            className="confirm-footer"
        >
            <div
                className="confirm-btn-sure"
                onClick={[Function]}
            >
                ok
            </div>
            <div
                className="confirm-btn-cancel"
                onClick={[Function]}
            >
                cancel
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="confirm"
>
      <div
            className="confirm-dialog"
      >
            <div
                  className="confirm-addon"
            >
                  <i
                        className="fa fa-exclamation-circle text-danger"
                  />
            </div>
            <div
                  className="confirm-body"
            >
                  Confirm test
            </div>
            <div
                  className="confirm-footer"
            >
                  <div
                        className="confirm-btn-sure"
                        onClick={[Function]}
                  >
                        ok
                  </div>
                  <div
                        className="confirm-btn-cancel"
                        onClick={[Function]}
                  >
                        cancel
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Confirm
        cancelText="cancel"
        okText="ok"
        onCancelClick={[Function]}
        onOkClick={[Function]}
        type="danger"
>
        Confirm test
</Confirm>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Confirm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "getAddonByType": [Function],
        "props": Object {
          "cancelText": "cancel",
          "children": "Confirm test",
          "okText": "ok",
          "onCancelClick": [Function],
          "onOkClick": [Function],
          "type": "danger",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="confirm"
>
          <div
                    className="confirm-dialog"
          >
                    <div
                              className="confirm-addon"
                    >
                              <i
                                        className="fa fa-exclamation-circle text-danger"
                              />
                    </div>
                    <div
                              className="confirm-body"
                    >
                              Confirm test
                    </div>
                    <div
                              className="confirm-footer"
                    >
                              <div
                                        className="confirm-btn-sure"
                                        onClick={[Function]}
                              >
                                        ok
                              </div>
                              <div
                                        className="confirm-btn-cancel"
                                        onClick={[Function]}
                              >
                                        cancel
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="confirm"
>
          <div
                    className="confirm-dialog"
          >
                    <div
                              className="confirm-addon"
                    >
                              <i
                                        className="fa fa-exclamation-circle text-danger"
                              />
                    </div>
                    <div
                              className="confirm-body"
                    >
                              Confirm test
                    </div>
                    <div
                              className="confirm-footer"
                    >
                              <div
                                        className="confirm-btn-sure"
                                        onClick={[Function]}
                              >
                                        ok
                              </div>
                              <div
                                        className="confirm-btn-cancel"
                                        onClick={[Function]}
                              >
                                        cancel
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Confirm
    cancelText="cancel"
    okText="ok"
    onCancelClick={[Function]}
    onOkClick={[Function]}
    type="danger"
>
    Confirm test
</Confirm>,
}
`;
